### PR TITLE
Enhancement/template generator

### LINKED ISSUE #21: use gulp to pre generated html files
we currently have 3 basically identical pages. should use a template engine to allow us to only need to maintain one page. would also mean processing the event.json at this time so the only client side js would be the countdown.

https://github.com/DanielBrierton/gulp-minibars

### PR DESCRIPTION
**Pull request by butlerx**
close #21

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +0,0 @@
-language: node_js
```
Comment (VoyTechnology) [inline .travis.yml]: I believe this change is being made in #58

File: README.md [hunk 2]
```diff
@@ -1,34 +1,37 @@
-# TechWeek Website [![Build Status](https://travis-ci.org/redbrick/techweek.dcu.ie.svg?branch=master)](https://travis-ci.org/redbrick/techweek.dcu.ie)
+# TechWeek Website [![CircleCI](https://circleci.com/gh/redbrick/techweek.dcu.ie/tree/master.svg?style=svg)](https://circleci.com/gh/redbrick/techweek.dcu.ie/tree/master)
```
Comment (VoyTechnology) [inline README.md]: Can you move it to another PR? This should be added in #58 but I didn't realize it at the time.

File: bin/generate.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+const Handlebars = require('handlebars');
+const fs = require('fs');
+const events = require(__dirname + '/../events.json');
+const template = fs.readFileSync(__dirname + '/../template.handlebars',  "utf-8");
+
+function render (source, events, url) {
+  let template = Handlebars.compile(source);
+  let output = template(events);
+  if (!fs.existsSync(__dirname + '/../dist')) { fs.mkdirSync(__dirname + '/../dist'); }
```
Comment (VoyTechnology) [inline bin/generate.js]: Can you move `fs.mkdirSync` to a new line?

File: bin/generate.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+const Handlebars = require('handlebars');
+const fs = require('fs');
+const events = require(__dirname + '/../events.json');
+const template = fs.readFileSync(__dirname + '/../template.handlebars',  "utf-8");
+
+function render (source, events, url) {
+  let template = Handlebars.compile(source);
+  let output = template(events);
```
Comment (VoyTechnology) [inline bin/generate.js]: I think inlining this would be neater as `template` its not used anywhere else.

`const output =  (Handlebars.compile(source))(events);`

File: bin/generate.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+const Handlebars = require('handlebars');
+const fs = require('fs');
+const events = require(__dirname + '/../events.json');
+const template = fs.readFileSync(__dirname + '/../template.handlebars',  "utf-8");
+
+function render (source, events, url) {
```
Comment (VoyTechnology) [inline bin/generate.js]: Can you add documentation to this function? What do `source`, `events` and `url` mean in this case.

File: bin/generate.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  let template = Handlebars.compile(source);
+  let output = template(events);
+  if (!fs.existsSync(__dirname + '/../dist')) { fs.mkdirSync(__dirname + '/../dist'); }
+  dir = __dirname + '/../dist';
```
Comment (VoyTechnology) [inline bin/generate.js]: Where is `dir` defined?

File: bin/generate.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  if (url) {
+    dir = dir + '/' + url;
```
Comment (VoyTechnology) [inline bin/generate.js]: Please use `path.join`.

File: bin/generate.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    if (!fs.existsSync(dir)){
+      fs.mkdirSync(dir);
+    }
+  }
+  fs.writeFile(dir + '/index.html', output, 'utf8', (err) => { if (err) throw err; });
```
Comment (VoyTechnology) [inline bin/generate.js]: Move `if` statement to a new line?

File: circle.yml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+machine:
```
Comment (VoyTechnology) [inline circle.yml]: This was added in #58. Please remove.

File: events/2014.json [hunk 10]
```diff
@@ -0,0 +1,194 @@
+{
+  "name": "TechWeek",
+  "location": "Dublin City University",
+  "start": 3,
+  "end": 7,
+  "month": "March",
+  "year": 2014,
+  "live": "2014-03-03T12:00:00",
```
Comment (VoyTechnology) [inline events/2014.json]: Would Unix timestamp be more appropiate? I am not sure is this representation understandable by languages other than JS.

File: gulpfile.js [hunk 11]
```diff
@@ -35,18 +36,19 @@ gulp.task('scss', function() {
...
     host: 'techweek.dev',
     livereload: true
   });
 });
 
 
 gulp.task('fonts', function() {
-   return gulp.src('./node_modules/materialize-css/fonts/**')
-       .pipe(gulp.dest('dist/fonts'));
+  return gulp.src('./node_modules/materialize-css/fonts/**')
```
Comment (VoyTechnology) [inline gulpfile.js]: I should probably make a new issue for this, would moving to https://github.com/material-components/material-components-web be beneficial seeing as its the (yet another) official Material Design framework?

Comment (butlerx) [inline gulpfile.js]: create a new issue for that, would also be good to remove jquery as part of that

File: events/2014.json (hunk 10 above)
Comment (butlerx) [inline events/2014.json]: id prefer the timestamp to be human readable

File: bin/generate.js (hunk 8 above)
Comment (butlerx) [inline bin/generate.js]: considering its an error check id prefer it to be one line

File: bin/generate.js (hunk 3 above)
Comment (butlerx) [inline bin/generate.js]: yes but id rather not

File: bin/generate.js (hunk 4 above)
Comment (butlerx) [inline bin/generate.js]: const would break it

File: gulpfile.js (hunk 11 above)
Comment (butlerx) [inline gulpfile.js]: issue #60 oppened for this

File: events.json [hunk 12]
```diff
@@ -1,183 +0,0 @@
-[{
```
Comment (SeanHealy33) [inline events.json]: I'm starting to think the json of the site should probably be in a separate repo since it adds a decent bloat to the site.

File: template.handlebars [hunk 13]
```diff
@@ -0,0 +1,74 @@
+<!DOCTYPE html>
```
Comment (SeanHealy33) [inline template.handlebars]: 😍 nice refactor 👍

File: events.json (hunk 12 above)
Comment (butlerx) [inline events.json]: yeah the json was originally never really meant to be part of the repo so moving it to a submoule would proabbly be a good idea
